metrics.go

```diff
@@ -3,47 +3,119 @@
 package goa
 
 import (
+	"regexp"
+	"strings"
+	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/armon/go-metrics"
 )
 
-// metriks is the local instance of metrics.Metrics
-var metriks *metrics.Metrics
+const (
+	allMatcher      string = "*/*"
+	allReplacement  string = "all"
+	normalizedToken string = "_"
+)
+
+var (
+	// metriks atomic value storage
+	metriks atomic.Value
+
+	// mu mutex for metriks
+	mu sync.Mutex
+
+	// used for normalizing names by matching '*' and '/' so they can be replaced.
+	invalidCharactersRE = regexp.MustCompile(`[\*/]`)
+
+	// Taken from https://github.com/prometheus/client_golang/blob/66058aac3a83021948e5fb12f1f408ff556b9037/prometheus/desc.go
+	metricsNameRE = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_:]*$`)
```

Comment:
I'm confused, it is actually used in normalize. How were you expecting the normalize function to operate? Currently this is it's flow.

```
normalizeKeys(keys):
    for each key in keys:
         if not validCharsRegex.Match(key):
               key = invalidCharsRegex.Replace(key, "_")
```